lib/ohai/plugins/hostname.rb

```diff
@@ -175,7 +175,18 @@ def collect_hostname
     else
       #host is not in dns. optionally use:
       #C:\WINDOWS\system32\drivers\etc\hosts
-      fqdn Socket.gethostbyaddr(info.last).first
+      found_fqdn = false
+      info[3..info.length].reverse.each do |addr|
+        hostent = Socket.gethostbyaddr(addr)
+        if hostent.first =~ /.+?\.(.*)/
+          fqdn hostent.first
+          found_fqdn = true
+          break
+        end
+      end
+      if !found_fqdn
```

Comment:
can this just be `fqdn info.first unless fqdn` and eliminate the `found_fqdn` variable entirely?  (i don't know if bare `fqdn` is a getter or not in this context...)